Alter status controller to allow any enclosed service to synchronously reply ok

mykh74 opened the pull request:
#7 alter status controller to allow any enclosed service to synchronously reply OK

mykh74 committed:
Merge pull request #1 from ecentria/master

New base from EcentriaRestBundle

Commit:
Alter status controller to allow any enclosed service to synchronously reply OK

jshanman commented on Controller/StatusController.php:
```diff
@@ -36,8 +40,14 @@ class StatusController extends FOSRestController
+        $event = new StatusCheckEvent();
+        $dispatcher->dispatch(Events::STATUS_CHECK, $event);
+        $messages = $event->getMessages();
+        if (count($messages)) {
```
we should add a "state" property to the event and check that instead of counting the messages.  Default state can be "ok" and can be set to "warning" or "error" by any listener.  A listener should not be able to "downgrade" state from "error" to "warning" so the setState method should only allow escalating the state and not downgrading it.

Commit:
added 'status' and 'exceptions' properties to StatusCheckEvent class

Commit:
added logic to prevent state downgrading in StatusCheckEvent

Commit:
- Adjusted the 2nd value in the response with nagios standard for errors: http://www.kernel-panic.it/openbsd/nagios/nagios6.html
- HTTP response code returns 200 when status state is in  0 (OK) or  1 (Warning), but 500 if state is 2 (Failure)
- fixed stack output

Commit:
when systems are normal, response returns empty array instead of string message

Commit:
when systems are normal, response returns empty array instead of string message

Commit:
Merge remote-tracking branch 'origin/Alter-status-controller-to-allow-any-enclosed-service-to-synchronously-reply-OK' into Alter-status-controller-to-allow-any-enclosed-service-to-synchronously-reply-OK

ggnet commented:
Looks good to me. I think we can merge it
cc @jshanman
